docs/execution-providers/CoreML-ExecutionProvider.md

```diff
@@ -49,18 +60,64 @@ Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_CoreML(so, coreml_fla
 Ort::Session session(env, model_path, so);
 ```
 
-## Configuration Options
+## Configuration Options (NEW API)
 
 There are several run time options available for the CoreML EP.
 
 To use the CoreML EP run time options, create an unsigned integer representing the options, and set each individual option by using the bitwise OR operator.
 
+ProviderOptions can be set by passing the unsigned integer to the `AppendExecutionProvider` method.
+```c++
+std::unordered_map<std::string, std::string> provider_options;
+provider_options["ModelFormat"] = std::to_string("MLProgram");
+provider_options["MLComputeUnits"] = std::to_string("ALL");
+provider_options["RequireStaticInputShapes"] = std::to_string("0");
+provider_options["EnableOnSubgraphs"] = std::to_string("0");
+```
+
+Python inference example code to use the CoreML EP run time options:
+```python
+import onnxruntime as ort
+model_path = "model.onnx"
+providers = [
+    ('CoreMLExecutionProvider', {
+        "ModelFormat": "MLProgram", "MLComputeUnits": "ALL", 
+        "RequireStaticInputShapes": "0", "EnableOnSubgraphs": "0"
+    }),
+]
+
+session = ort.InferenceSession(model_path, providers=providers)
+outputs = ort_sess.run(None, input_feed)
+```
+
+### Available Options (New API)
+`ModelFormat` can be one of the following values:
+- `MLProgram`: Create an MLProgram format model. Requires Core ML 5 or later (iOS 15+ or macOS 12+).
+- `NeuralNetwork`: Create a NeuralNetwork format model. Requires Core ML 3 or later (iOS 13+ or macOS 10.15+).
+
+`MLComputeUnits` can be one of the following values:
+- `CPUOnly`: Limit CoreML to running on CPU only.
+- `CPUAndNeuralEngine`: Enable CoreML EP for Apple devices with a compatible Apple Neural Engine (ANE).
+- `CPUAndGPU`: Enable CoreML EP for Apple devices with a compatible GPU.
+- `ALL`: Enable CoreML EP for all compatible Apple devices.
+
+`RequireStaticInputShapes` can be one of the following values:
+- `0`: Allow the CoreML EP to take nodes with inputs that have dynamic shapes.
```

Comment:
nit: would be good to note that performance may be affected. could maybe copy text from the COREML_FLAG_ONLY_ALLOW_STATIC_INPUT_SHAPES description.

Reply:
Sure.